js/modules/k6/experimental/streams/module.go

```diff
@@ -0,0 +1,285 @@
+		}
+
+		// Then we try to convert it to an UnderlyingSource
+		underlyingSource = call.Arguments[0].ToObject(rt)
+		underlyingSourceDict, err = NewUnderlyingSourceFromObject(rt, underlyingSource)
+		if err != nil {
+			throw(rt, err)
+		}
+	}
+
+	// 3. Perform ! InitializeReadableStream(this).
+	stream := &ReadableStream{
+		runtime: mi.vu.Runtime(),
+		vu:      mi.vu,
+	}
+	stream.initialize()
+
+	// 4. If underlyingSourceDict["type"] is "bytes":
+	if underlyingSourceDict.Type == "bytes" {
+		common.Throw(stream.runtime, newError(RuntimeError, "'bytes' stream is not supported yet"))
+	} else { // 5. Otherwise,
+		// 5.1. Assert: underlyingSourceDict["type"] does not exist.
+		if underlyingSourceDict.Type != "" {
+			common.Throw(rt, newError(AssertionError, "type must not be set for non-byte streams"))
+		}
+
+		// 5.2. Let sizeAlgorithm be ! ExtractSizeAlgorithm(strategy).
+		sizeAlgorithm := extractSizeAlgorithm(rt, strategy)
+
+		// 5.3. Let highWaterMark be ? ExtractHighWaterMark(strategy, 1).
+		highWaterMark := extractHighWaterMark(rt, strategy, 1)
+
+		// 5.4. Perform ? SetUpReadableStreamDefaultControllerFromUnderlyingSource(...).
+		stream.setupReadableStreamDefaultControllerFromUnderlyingSource(
+			underlyingSource,
+			underlyingSourceDict,
+			highWaterMark,
+			sizeAlgorithm,
+		)
+	}
+
+	streamObj := rt.ToValue(stream).ToObject(rt)
+
+	proto := call.This.Prototype()
+	if proto.Get("locked") == nil {
+		err = proto.DefineAccessorProperty("locked", rt.ToValue(func() goja.Value {
+			return rt.ToValue(stream.Locked)
+		}), nil, goja.FLAG_FALSE, goja.FLAG_TRUE)
+		if err != nil {
+			common.Throw(rt, newError(RuntimeError, err.Error()))
+		}
+	}
+
+	err = streamObj.SetPrototype(proto)
+	if err != nil {
+		common.Throw(rt, newError(RuntimeError, err.Error()))
+	}
+
+	return streamObj
+}
+func defaultSizeFunc(_ goja.Value) (float64, error) { return 1.0, nil }
+
+func (mi *ModuleInstance) initializeStrategy(call goja.ConstructorCall) *goja.Object {
+	runtime := mi.vu.Runtime()
+
+	// Either if the strategy is not provided or if it doesn't have a 'highWaterMark',
+	// we need to set its default value (highWaterMark=1).
+	// https://streams.spec.whatwg.org/#rs-prototype
+	strArg := runtime.NewObject()
+	if len(call.Arguments) > 1 && !common.IsNullish(call.Arguments[1]) {
+		strArg = call.Arguments[1].ToObject(runtime)
+	}
+	if common.IsNullish(strArg.Get("highWaterMark")) {
+		if err := strArg.Set("highWaterMark", runtime.ToValue(1)); err != nil {
+			common.Throw(runtime, newError(RuntimeError, err.Error()))
+		}
+	}
+
+	// If the stream type is 'bytes', we don't want the size function.
+	// Except, when it is manually specified.
+	size := runtime.ToValue(defaultSizeFunc)
+	if len(call.Arguments) > 0 && !common.IsNullish(call.Arguments[0]) {
+		srcArg := call.Arguments[0].ToObject(runtime)
+		if !common.IsNullish(srcArg.Get("type")) && srcArg.Get("type").String() == ReadableStreamTypeBytes {
+			size = nil
```

Comment:
in general `.Get` isn't particularly ... performant. So we might prefer to not call it multiple times with the same arguments unless the specification requires it.

edit: I will not be making the same comment on each place this happens